### DISCUSSION TITLE
Running gdUnit4Net tests without launching Godot instance

### DISCUSSION THREAD

**Discussion by dvankley** [MikeSchulze/gdUnit4Mono#151] [Q&A] [answered]
Thanks for your work on this tool, it's very useful.

Is there a way to launch gdUnit4Net tests without the test run automatically launching an instance of Godot? I ask because I have internal logic not coupled to Godot I'd like to test without the performance overhead.

I've tried separate .NET projects for pure unit tests and gdUnit4Net integration tests and have had issues with that. If that's the preferred approach I can dig further into that.

**Comment by MikeSchulze** [comment]
I also thought about whether it is really necessary to always use Godot as an execution program for tests.
At the moment this is not possible, there will always be a scene running to execute the test execution logic.

From a unit test perspective, it should not be necessary to use the Godot engine as long as no core functions are used that have runtime dependencies.
However, it would be a problem if a tester wanted to test Godot signals or the initial state of nodes when adding them to a scene.

Therefore, a stricter distinction would have to be made between unit and integration tests.
This means that a unit test can never test signals or use scenes.

To achieve this, a test would first have to be scanned to determine whether it uses such code paths in order to decide whether it is a unit or integration test.
This would be very time-consuming, and support for test categories would probably be the best and most sensible solution here.

There is an open feature request to support test filters, https://github.com/MikeSchulze/gdUnit4Net/issues/27
this will be the right direction.

I can do a prototype in the next week to verify I can run unit tests without using Godot engine, by using test category `Unit` and run in a Godot context when is set to `Integration`.

**Comment by dvankley** [comment]
>I can do a prototype in the next week to verify I can run unit tests without using Godot engine, by using test category Unit and run in a Godot context when is set to Integration.

Sounds great, thanks!

**Reply by MikeSchulze** [reply]
I open a feature request https://github.com/MikeSchulze/gdUnit4Net/issues/152

You can vote on https://discord.com/channels/885149082119733269/1113541429134905404/1302308189819371611

**Comment by MikeSchulze** [comment, marked as answer]
With GdUnit4Net v5.0.0 tests without `[RequireGodotRunime]` will now execute without starting a Godot instance.
